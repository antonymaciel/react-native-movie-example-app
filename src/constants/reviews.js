export const REVIEWS_SUCCESS = 'REVIEWS_SUCCESS';
export const REVIEWS_FAILD = 'REVIEWS_FAILD';
export const REVIEWS_LOADING = 'REVIEWS_LOADING';
export const REVIEWS_FINISH = 'REVIEWS_FINISH';
