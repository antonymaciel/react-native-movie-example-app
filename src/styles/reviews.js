import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
    title: {
        fontSize: 20,
        fontWeight: '500'
    },
    headerContainer: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        margin: 20,
    },
    image: {
        marginRight: 40,
        height: 130,
        width: 90
    },
    reviewList: {
        flex: 1
    }
});


export default styles;
