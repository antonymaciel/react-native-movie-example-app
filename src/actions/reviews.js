import Api from "../api/apiMovies"
import { REVIEWS_SUCCESS, REVIEWS_FAILD, REVIEWS_LOADING, REVIEWS_FINISH } from '../constants/reviews'

export const movieReviewsSuccess = (reviews) => ({
    type: REVIEWS_SUCCESS,
    reviews
});

export const movieReviewsFaild = () => ({
    type: REVIEWS_FAILD, 
});

export const movieReviewsLoading = () => ({
    type: REVIEWS_LOADING,
});

export const movieReviewsFinish = () => ({
    type: REVIEWS_FINISH,
});

export const getMovieReviews = (idMovie, page) => (dispatch) => {
    dispatch(movieReviewsLoading());
    Api.getMovieReviews(idMovie, page)
    .then((json) => {
        dispatch(movieReviewsSuccess(json));
    })  
    .catch(err => {
        console.log(err);
        dispatch(movieReviewsFaild());
    })
  };

