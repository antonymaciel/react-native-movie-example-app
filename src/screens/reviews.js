import React from "react";
import { View, Text } from "react-native"
import styles from "../styles";

function Reviews(props) {
    return (<View style={styles.pageContainer}><Text>{props.msg}</Text></View>);
}

export default Reviews;