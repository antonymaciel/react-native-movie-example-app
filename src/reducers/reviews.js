import { REVIEWS_SUCCESS, REVIEWS_FAILD, REVIEWS_LOADING, REVIEWS_FINISH } from '../constants/reviews'

export const initialState = {
    reviews: null,
    totalPages: null, 
    totalReviews: null,
    loading: false,
    error: false
}


const reviews = (state = initialState, action) => {
    switch(action.type){
        case REVIEWS_SUCCESS:
            return {
                reviews: state.reviews ? [...state.reviews.results, ...action.reviews.results] : action.reviews.results,
                totalReviews: action.reviews.total_results,
                totalPages: action.reviews.total_pages,
                loading: false, 
                error: false
            };
        case REVIEWS_LOADING:
            return { 
                ...state, 
                loading: true,
                error: false
            };
        case REVIEWS_FAILD:
            return { 
                ...state, 
                loading: false, 
                error: true
            };
        case REVIEWS_FINISH:
            return initialState;    
        default:
                return state;
    } 
}

export default reviews;