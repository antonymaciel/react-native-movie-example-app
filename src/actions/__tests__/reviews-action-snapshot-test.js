import mockStore from 'redux-mock-store';
import { getMovieReviews } from '../reviews';
jest.mock('../../api/apiMovies');

it('should handle reviews success action', async () => {
    const store = mockStore();
    await store.dispatch(getMovieReviews(0, 1));
    expect(store.getActions()).toMatchSnapshot();
});
