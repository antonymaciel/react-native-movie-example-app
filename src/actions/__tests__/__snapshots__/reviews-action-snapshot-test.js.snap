// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle reviews success action 1`] = `
Array [
  Object {
    "type": "REVIEWS_LOADING",
  },
  Object {
    "reviews": Array [
      Object {
        "movie": 0,
        "results": Array [
          Object {
            "author": "Mark",
          },
          Object {
            "author": "Paul",
          },
        ],
      },
    ],
    "type": "REVIEWS_SUCCESS",
  },
]
`;
